source/common/upstream/cluster_manager_impl.cc

```diff
@@ -154,6 +154,38 @@ void ClusterManagerInitHelper::setInitializedCb(std::function<void()> callback)
   }
 }
 
+void ClusterManagerImpl::initializeClustersFromV1Json(const Json::Object& config) {
+  envoy::api::v2::ConfigSource cds_config;
+  if (config.hasObject("cds")) {
+    envoy::api::v2::Cluster cds_cluster;
+    Config::CdsJson::translateCluster(*config.getObject("cds")->getObject("cluster"), sds_config_,
+                                      cds_cluster);
+    loadCluster(cds_cluster, false);
+    Config::Utility::translateCdsConfig(*config.getObject("cds"), cds_config);
+    // We can now potentially create the CDS API once the backing cluster exists.
+    cds_api_ = factory_.createCds(cds_config, sds_config_, *this);
+    init_helper_.setCds(cds_api_.get());
+  } else {
+    init_helper_.setCds(nullptr);
+  }
+
+  for (const Json::ObjectSharedPtr& cluster_config : config.getObjectArray("clusters")) {
+    envoy::api::v2::Cluster cluster;
+    Config::CdsJson::translateCluster(*cluster_config, sds_config_, cluster);
+    loadCluster(cluster, false);
+  }
+}
+
+void ClusterManagerImpl::initializeClustersFromV2Proto(const envoy::api::v2::Bootstrap& bootstrap) {
+  for (const auto& cluster : bootstrap.bootstrap_clusters()) {
+    loadCluster(cluster, false);
+  }
+
+  // We can now potentially create the CDS API once the backing cluster exists.
```

Comment:
This is one of those things that is not explicitly required but I guarantee will be a source of griping. IMO given how relatively easy to fix it is, I think we should just do it.